lib/googleauth/user_authorizer.rb

```diff
@@ -228,6 +248,23 @@ def store_credentials user_id, credentials
         credentials
       end
 
+      # The code verifier for PKCE for OAuth 2.0. When set, the
+      # authorization URI will contain the Code Challenge and Code
+      # Challenge Method querystring parameters, and the token URI will
+      # contain the Code Verifier parameter.
+      #
+      # @param [String|nil] new_code_erifier
+      def code_verifier= new_code_verifier
+        @code_verifier = new_code_verifier
+      end
+
+      # Generate the code verifier needed to be sent while fetching
+      # authorization URL.
+      def generate_code_verifier
+        random_number = rand 32..96
+        SecureRandom.alphanumeric(random_number).to_str
```

Comment:
I think `SecureRandom.alphanumeric` returns a string. Do we need to call `to_str` on the result?

Reply:
No, we don't. Missed that.